src/main/java/io/vertx/ext/dropwizard/impl/Matcher.java

```diff
@@ -3,43 +3,60 @@
 import io.vertx.ext.dropwizard.Match;
 import io.vertx.ext.dropwizard.MatchType;
 
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
+import java.util.Map.Entry;
 import java.util.regex.Pattern;
 
 /**
  * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
  */
 class Matcher {
-
   private final Set<String> equalsMatches;
-  private final Pattern[] regexMatches;
+  private final Map<Pattern, String> regexMatches;
+  private final Map<String, String> identifiers;
 
   Matcher(List<Match> matches) {
     equalsMatches = new HashSet<>();
+    regexMatches = new HashMap<>();
+    identifiers = new HashMap<>();
+
     for (Match match : matches) {
       if (match.getType() == MatchType.EQUALS && match.getValue() != null) {
         equalsMatches.add(match.getValue());
+      } else if (match.getType() == MatchType.REGEX && match.getValue() != null) {
+        regexMatches.put(Pattern.compile(match.getValue()), match.getValue());
+      }
+
+      if (match.getIdentifier() != null) {
+        identifiers.put(match.getValue(), match.getIdentifier());
       }
     }
-    regexMatches = matches.stream().
-        filter(matcher -> matcher.getType() == MatchType.REGEX && matcher.getValue() != null).
-        map(matcher -> Pattern.compile(matcher.getValue())).
-        toArray(Pattern[]::new);
   }
 
-  boolean match(String value) {
+  String match(String value) {
     if (equalsMatches.size() > 0 && equalsMatches.contains(value)) {
-      return true;
+      return value;
     }
-    if (regexMatches.length > 0) {
-      for (Pattern pattern : regexMatches) {
-        if (pattern.matcher(value).matches()) {
-          return true;
+
+    if (regexMatches.size() > 0) {
+      for (Entry<Pattern, String> entry : regexMatches.entrySet()) {
```

Comment:
should void this as it creates an iterator, instead we should have the matches in an array and avoid using iterators() there